pkg/kubectl/cmd/util/factory.go

```diff
@@ -162,11 +162,11 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 				}
 				return kubectl.MakeLabels(t.Spec.Selector), nil
 			default:
-				kind, err := meta.NewAccessor().Kind(object)
+				_, kind, err := api.Scheme.ObjectVersionAndKind(object)
```

Comment:
Ah, this is what you were asking me about. It's OK for now because unversioned object kinds all match the versioned object kinds. Hopefully we'll have a versioned client (#4874) before this becomes a problem.